packages/app/src/runner/studio/AssertionOptions.ce.vue

```diff
@@ -0,0 +1,92 @@
+<template>
+  <div
+    ref="popper"
+    class="assertion-options"
+  >
+    <div
+      v-for="{ name, value } in options"
+      :key="`${name}${value}`"
+      class="assertion-option"
+      @click.stop="() => onClick(name, value)"
+    >
+      <span
+        v-if="name"
+        class="assertion-option-name"
+      >
+        {{ truncate(name) }}:{{ ' ' }}
+      </span>
+      <span
+        v-else
+        class="assertion-option-value"
+      >
+        {{ truncate(value) }}
+      </span>
+    </div>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import { createPopper } from '@popperjs/core'
+import { onMounted, ref, nextTick, Ref } from 'vue'
+
+const props = defineProps<{
+  type: string
+  addAssertion: any
+  options: any
+  setPopperElement: any
+}>()
+
+const truncate = (str) => {
+  if (str && str.length > 80) {
+    return `${str.substr(0, 77)}...`
+  }
+
+  return str
+}
+
+const popper: Ref<HTMLElement | null> = ref(null)
+
+onMounted(() => {
+  nextTick(() => {
+    const popperEl = popper.value as HTMLElement
```

Comment:
Is this cast necessary? It shouldn't be?

If it's the null check you are concerned with, I thin we should be throwing an error here. If it's null, we end up doing `(popperEl.parentElement`) which will be `undefined.parentElement`, which is far harder to debug than a more explicit "XXX element was not found. This should be impossible".

Reply:
`ref`s are set to `null` in the setup because we need an initial value. But when they're set to the elements in the `template`, they'll never be changed. So I used `cast`. I think it's really unlikely for them to be changed back to `null`.